index.js

```diff
@@ -73,28 +91,29 @@ nlf.find({ directory, production }, (err, npmPackages) => {
       builders.forEach(builder => {
         builder.addCached(name, version, lastVersionDefinition);
       });
-    } else if (_.get(npmPackage, 'licenseSources.package.sources', []).length > 0) {
-      const packageSources = _.get(npmPackage, 'licenseSources.package.sources', [])
-        .map(({ license }) => license)
-        .join(', ');
+    } else if (_.get(npmPackage, 'license', '').length > 0) {
+      const packagelicenseFile = `${_.get(npmPackage, 'license')}`;
 
       console.log(
-        chalk.red(`License source not found for ${packageID}, derived license types are: ${packageSources}.`)
+        chalk.red(`License source not found for ${packageID}, derived license types are: ${packagelicenseFile}.`)
       );
 
       builders.forEach(builder => {
         builder.addEmpty(name, version);
-      });
+      }); 
     } else {
       console.log(chalk.red(`License could not be detected for package ${packageID}.`));
 
       builders.forEach(builder => {
         builder.addEmpty(name, version);
       });
     }
+    promises.push(builders[1].buildLink(name, npmPackage.repository, npmPackage.licenseFile));
```

Comment:
May consider assigning the JSON builder to it's own variable so it can be used here explicitly. E.g.

```
const jsonBuilder = new JSONBuilder();
const builders = [new TextBuilder(), jsonBuilder]
```

This way, even if someone rearranges the builders, since nowhere else does order matter, it won't break this.

Reply:
Yeah, good idea. meant to change this, will fix